### ISSUE TITLE
Are the private groups tests working as intended?

### ISSUE THREAD

**Issue by davehunt** [mozilla/mozillians-tests#221]
I've recently looked into creating new test accounts, and the 'private' account has me concerned that we're not testing what we mean to. The tests `test_private_groups_field_as_public_when_logged_in` and `test_private_groups_field_when_not_logged_in` use the 'private' user to check that the groups are not displayed to the public. As far as I can tell, this user is not vouched on staging or production, and does not belong to any groups. I can't see a way that the groups would ever be displayed for this user, regardless of the privacy settings. I think these tests are giving false positives and should either be deleted (along with the private account) or should be fixed.

/cc @johngian @akatsoulas

**Comment by stephendonner** [comment]
Also /cc @comzeradd

**Comment by johngian** [comment]
Regarding ``test_private_groups_field_as_public_when_logged_in``:
* In mozillians.org profiles we have a functionality to allow users to view their profile as a different user (eg. to see how the profile gets rendered to non logged-in users). The idea here is that for a private field, specifically ``groups`` which should be visible to the profile owner, when the user tries to see the profile as ``Public`` this field should not be rendered.

Regarding ``test_private_groups_field_when_not_logged_in`` I think the idea is to ensure that private fields are not rendered to unauthenticated users.

In both tests, in order to avoid false positives I think we need to vouch user ``private_user`` and add this user to some groups. Also the group privacy field should be ``Mozillians``.

**Comment by davehunt** [comment]
I agree with all your comments @johngian with the addition of adding checks for the adverse expectation. By that I mean the property that returns if the groups are displayed should somewhere be checked for a positive return value.

**Comment by johngian** [comment]
Sounds good :+1:
